ArtemisKit/Sources/Messages/ViewModels/SendMessageViewModels/FileUpload/SendMessageUploadFileViewModel.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  SendMessageUploadFileViewModel.swift
+//  ArtemisKit
+//
+//  Created by Eylul Naz Can on 9.12.2024.
+//
+
+import Foundation
+import UniformTypeIdentifiers
+import Common
+import SwiftUI
+
+@Observable
+final class SendMessageUploadFileViewModel: UploadViewModel {
+
+    var fileData: Data?
+    var fileName: String?
+    var isFilePickerPresented = false
+
+    private let messagesService: MessagesService
+
+    static let allowedFileTypes: [UTType] = [
+        .png,
+        .jpeg,
+        .gif,
+        .svg,
+        .pdf,
+        .rtf,
+        .plainText,
+        .json,
+        .spreadsheet,
+        .presentation,
+        UTType(filenameExtension: "doc") ?? .data,
+        UTType(filenameExtension: "docx") ?? .data,
+        UTType(filenameExtension: "xls") ?? .spreadsheet,
+        UTType(filenameExtension: "xlsx") ?? .spreadsheet,
+        UTType(filenameExtension: "ppt") ?? .presentation,
+        UTType(filenameExtension: "pptx") ?? .presentation
+    ]
+
+    init(
+        courseId: Int,
+        conversationId: Int64,
+        messagesService: MessagesService = MessagesServiceFactory.shared
+    ) {
+        self.messagesService = messagesService
+        super.init(courseId: courseId, conversationId: conversationId)
+    }
+
+    /// Handles changes to the selected file URL
+    func filePicked(at url: URL) {
+        isFilePickerPresented = false
+        loadFileData(from: url)
+    }
+
+    /// Reads the file data from the provided URL
+    private func loadFileData(from url: URL) {
+        uploadState = .compressing
+        filePath = nil
+
+        Task {
```

Comment:
I don't think we need to wrap this block inside a `Task` (we don't use async/await here).